### PR TITLE
Call mavlink_parse_char without r_msg/r_status

### PR DESCRIPTION
**Pull request by peterbarker**
We don't use the values that are returned inside these two objects, but passing them into the mavlink API causes a chunk of work to be done (particularly filling `r_status` in for every character that's passed in.

Also included is an example which calls the API with/without objects.  It returns this on a `Pixhaw1-1M` here:

```
Decode rate    (with return-objects): 494354 bytes/s
dec=2228224 t=2869610
Decode rate (without return-objects): 776490 bytes/s
dec=2228224 t=4509092
Decode rate    (with return-objects): 494162 bytes/s
dec=2228224 t=2870253
Decode rate (without return-objects): 776316 bytes/s
dec=2228224 t=4509270
Decode rate    (with return-objects): 494142 bytes/s
dec=2228224 t=2870199
Decode rate (without return-objects): 776330 bytes/s
dec=2228224 t=4509244
Decode rate    (with return-objects): 494145 bytes/s
dec=2228224 t=2870286
```

That's a signficant change.

### CODE REVIEW

**Comment by tridge** [comment]
need to test:
 - signing, both with correct key and incorrect key
 - MAV statistics with some deliberate lost packets, ensure statistics are OK in log

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1487,9 +1483,11 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
         bool parsed_packet = false;
 
         // Try to get a new message
-        if (mavlink_parse_char(chan, c, &msg, &status)) {
-            hal.util->persistent_data.last_mavlink_msgid = msg.msgid;
-            packetReceived(status, msg);
+        if (mavlink_parse_char(chan, c, NULL, NULL)) {
```
Comment (khancyr) [inline libraries/GCS_MAVLink/GCS_Common.cpp:1486]: nullptr, otherwise static analyzer spam with warning

Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp:1486]: This is a call into the C library bindings.  static analyzer should see the markers in the relevant header and be good with it.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -1487,9 +1483,11 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
...
+            const mavlink_message_t *msg = GCS_MAVLINK::get_channel_buffer(chan);
+            const mavlink_status_t *status = GCS_MAVLINK::get_channel_status(chan);;
```
Comment (magicrub) [inline libraries/GCS_MAVLink/GCS_Common.cpp:1488]: extra ;

Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp:1488]: Fixed

**Comment by tridge** [comment]
@peterbarker happy for you to merge when you've confirmed MAV behaviour

**Comment by khancyr** [comment]
@peterbarker I think this should be great to merge. Can you rebase and make the latest check ?

**Pull request closed without merging**